### PR TITLE
Add Slack Source

### PR DESCRIPTION
**Pull request by c4urself**
This is a WFM version of a Slack source for Ibid. I've probably missed a bunch of things so point me in the right direction and I'll try to get that fixed.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,5 @@
 /local.ini
 /logs/
 dropin.cache
+docs/_build
+twistd.pid
```
Comment (stefanor) [inline .gitignore]: Both of those could probably use a leading `/`

File: ibid/plugins/factoid.py [hunk 2]
```diff
@@ -794,7 +794,7 @@ def modify(self, event, name, number, pattern, is_regex, operation, sep):
         u'hoezit', u'wotcha', u'wotcher', u'yo', u'word', u'good day',
         u'wasup', u'wassup', u'howzit', u'howsit', u'buon giorno',
         u'hoe lyk it', u'hoe gaan dit', u'good morning', u'morning',
-        u'afternoon', u'evening',
+        u'afternoon', u'evening', u'goedemorgen'
```
Comment (stefanor) [inline ibid/plugins/factoid.py]: Seems out of the scope of this PR, but not a problem.

Add a trailing `,`?

File: ibid/source/slack.py [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+        self.channels = []
+        self.groups = []
+        self.users = {}
+        self.user_id = None
+        self.name = ''
+
+    def clean_slack_msg(self, raw_msg):
+        """ For no good reason slack uses <@`uid`> instead of
+        `name`: so we have to clean it up a little
+        """
```
Comment (stefanor) [inline ibid/source/slack.py]: Should we try to handle this more centrally? I'm thinking of an API where the core call into sources, to find out how replies should be formatted.

Comment (c4urself) [inline ibid/source/slack.py]: sounds like a good idea -- do other platforms also need to clean the raw messages?

Comment (c4urself) [inline ibid/source/slack.py]: keep wanting to come back to this but don't think i'll have much time for it in the foreseeable future

**Comment by c4urself** [comment]
We used this daily for ~2 years at work -- seemed to work fine :) -- going to close now since there doesn't seem to be much interest for it and since we've moved on to new things at work as well.

**Pull request closed without merging**